cpp/src/slice2java/Gen.cpp

```diff
@@ -4448,3 +4326,109 @@ Slice::Gen::ProxyVisitor::visitOperation(const OperationPtr& p)
         out << eb;
     }
 }
+
+void
+Slice::Gen::TypesVisitor::emitCompactIdHelper(const ClassDefPtr& p)
+{
+    string prefix = getPackagePrefix(p);
+    if (!prefix.empty())
+    {
+        prefix = prefix + ".";
+    }
```

Comment:
we can move this inside the other if clause, is not used otherwise.

Reply:
Fixed as suggested.